docs/source/guide/build_pipelines/pipeline_testing.rst

```diff
@@ -0,0 +1,289 @@
+needs/contexts. For example, if the tested function's output is a NumPy array, then the user can replace the templated
+``self.assertEqual(sample_output_generated, sample_output_expected)`` with the customized ``assert all(sample_output_generated == sample_output_expected)``,
+which would result in proper equality evaluation.
+
+Running Pipeline Test
+---------------------
+
+With such limitations in templating, most test methods out of the box are bound to fail. However, to avoid any potential interference
+with the user's system, test failures are wrapped inside the ``try``-``except`` block, which makes all test methods run successfully
+without erroring out. Instead, the user is informed of failed test methods via warning messages that ask the user to check the code
+and adapt it if needed.
+
+Hence, for the ``iris_pipeline`` example discussed here, running
+
+.. code-block:: bash
+
+    cd ./output/pipeline_basics/
+    python -m unittest test_iris_pipeline.TestIrisPipeline
+
+would result in
+
+.. code-block:: none
+
+    /Users/sangyoonpark/Projects/Linea/prod/lineapy/examples/tutorials/output/pipeline_basics/test_iris_pipeline.py:108: UserWarning: Test failed, but this may be due to our limited templating. Please adapt the test as needed.
+    warnings.warn(
+    ./Users/sangyoonpark/Projects/Linea/prod/lineapy/examples/tutorials/output/pipeline_basics/test_iris_pipeline.py:79: UserWarning: Test failed, but this may be due to our limited templating. Please adapt the test as needed.
+    warnings.warn(
+    .
+    ----------------------------------------------------------------------
+    Ran 2 tests in 0.262s
+
+    OK
+
+.. warning::
+
+    Tests may involve long compute and/or large storage, so please take care in running them.
+
+Adapting Pipeline Test
+----------------------
+
+As shown, both test methods "failed" (i.e., warning messages raised) for ``iris_pipeline``. This is not surprising because,
+again, the scaffold is relying on a naive mode of equality evaluation via ``unittest``'s ``assertEqual()``, which does not
+work for more sophisticated object types such as ``pandas.DataFrame`` and ``sklearn.linear_model.LinearRegression``.
+For more proper equality evaluation, we may replace existing ``try``-``except`` blocks with new ``assert`` statements,
+like so (updates highlighted in yellow):
+
+.. code-block:: python
+   :emphasize-lines: 34, 56, 57, 58
+
+    # ./output/pipeline_basics/test_iris_pipeline.py
+
+    import os
+    import pickle
+    import unittest
+    import warnings
+    from pathlib import Path
+    from typing import Callable
+
+    from iris_pipeline_module import get_iris_model, get_iris_preprocessed
+
+    [...]
+
+    class TestIrisPipeline(unittest.TestCase):
+
+        [...]
+
+        def test_get_iris_preprocessed(self) -> None:
+            """
+            NOTE: The code below is provided as scaffolding/template.
```

Comment:
sometime you are using scaffolding sometime you are using scaffold ... (that's fine, just show you I've read)

Reply:
Edits made.